templates/repo/header.tmpl

```diff
@@ -5,14 +5,15 @@
 			<div class="ui huge breadcrumb repo-title">
 				{{if .RelAvatarLink}}
 				<img class="ui avatar image" src="{{.RelAvatarLink}}">
+				{{else if .IsTemplate}}
+					<svg class="mega-octicon-svg"><use xlink:href="{{StaticUrlPrefix}}/img/svg/octicons.svg#repo-template{{if .IsPrivate}}-private{{end}}" /></svg>
```

Comment:
Maybe a template helper would be nice, like `{{svg "mega-octicon" 30 30}}`. Arguments would be name,width,height. Width and heigth can default to a commonly used like 16px.

Reply:
Also note there is a existing rule in Less for `.repository .repo-header .mega-octicon` that can be removed with this.

Reply:
I am currently modifying them to be a template helper, but not with heigh/width parameters. imo those should be controlled through CSS like they currently are so that we don't start mixing code with styling.

`{{octicon <is-mega> <name>}}` is my current approach, e.g. `{{octicon false "repo"}}`  
Alternatively, I could do `{{octicon <name>}}` and `{{mega-octicon <name>}}`

The template helper is nice because it's much easier to read the underlying code once and know that the corresponding HTML will be correct.

For perspective, with the current solution I have been able to swap out almost half of the existing icons in our templates with no other changes required (unless they use `font-size` or some other combo of classes)

Reply:
Fine with me to keep sizing to CSS but "mega" only refers to size, right? I guess that parameter won't be needed then and the helper can take a string argument only.

I'd suggest defining a low-specificity selector that matches all SVG icons:

````css
svg {
 width: 16px;
 height: 16px;
 color: currentcolor;
}
````
This base of 16px size can then be overwritten via more specific selectors and it confines SVG that may come with different sizes in their attributes to our base size.

Note that casing of `currentcolor` is intended. This casing was changed in a recent CSS spec, thought it does not really matter as CSS is case-insensitive for values anyways.

Reply:
Currently I still use `mega` in the helper to ease transition, since most of our existing CSS rules don't specify size and rely on either having a class of `octicon` or `mega-octicon`